Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -474,13 +475,10 @@ void ACesium3DTileset::OnFocusEditorViewportOnThis() {
       *this->GetName());
 
   struct CalculateECEFCameraPosition {
```

Comment:
I don't think what you proposed will work with std::visit. I couldn't get it to work, at least. VC++ reports:

```
error C2672: 'std::visit': no matching overloaded function found
note: could be 'unknown-type std::visit(_Callable &&,_Variants &&...)'
note: 'unknown-type std::visit(_Callable &&,_Variants &&...)': could not deduce template argument for '_Callable'
```

I think the problem is that when we write `std::visit(CalculateECEFCameraPosition, boundingVolume)`, the compiler needs to be able to resolve `CalculateECEFCameraPosition` to a particular overload of the function. But that's not possible here because we need it to consider all the overloads when dispatching at runtime, not pick one at the call site.

The cppreference.com [page](https://en.cppreference.com/w/cpp/utility/variant/visit) isn't terribly easy to read, but I think it's presenting at the bottom an `overloaded` template that can be used to resolve this. But it's just a struct again. In C++20 it's apparently provided out of the box, fwiw.

In any case, I don't think there's anything inherently dodgy about a struct with no fields, so I'd rather just leave this as-is.

Reply:
Thanks for the explanation. I haven't personally implemented the visitor pattern using std variants and visit, so I did brush up on it little [here](https://www.cppstories.com/2018/06/variant/). This overload pattern look legitimate, if not really to my taste, from a language perspective. Looks like there are more instances of it in cesium-native as well.